src/core/experimental/array.d

```diff
@@ -23,6 +23,34 @@ private template ElementType(R)
         alias ElementType = void;
 }
 
+
+/**
+ * Create an empty (qualified) `rcarray`.
+ *
+ * Returns:
+ *      an empty `rcarray`
+ */
+auto make(Q : rcarray!T, T)(size_t initialCapacity = 0)
```

Comment:
This factory function is at the module level. The reason for this is that this way one is able to specify the mutability qualifier, without repeating the name of the struct (i.e. `auto ia = make!(immutable rcarray!int);` instead of `auto ia = rcarray!int.make!(immutable rcarray!int);`). This does pollute the module and complicates selective imports.

The alternative is to simply not allow construction of `const`/`immutable` arrays through a factory function. That way, the factory function can be moved into the type (and optionally renamed, giving e.g `auto a = rcarray!int.empty;`). This might be fine, as I can't think of many legitimate use cases of constructing empty `const`/`immutable` arrays anyway.

(Sadly, a factory function is necessary to be able to create empty `rcarray`s without having to explicitly specify the initial capacity. We can't use the default constructor, as that would leave `__RefCount`, which allocates memory in its constructors, uninitialised).

Reply:
We already have https://dlang.org/phobos/std_container_util.html#.make. I don't think it would be particularly confusing. That said, I still don't know if it's the best option.